js_src/_dart_helpers.js

```diff
@@ -131,6 +131,12 @@ function _createReactDartComponentClass2(dartInteropStatics, componentStatics, j
       return typeof derivedState !== 'undefined' ? derivedState : null;
     }
     render() {
+      // Must call checkPropTypes manually because React moved away from using the `prop-types` package.
+      // See: https://github.com/facebook/react/pull/18127
+      // React now uses its own internal cache of errors for PropTypes which broke `PropTypes.resetWarningCache()`.
+      // Solution was to use `PropTypes.checkPropTypes` directly which makes `PropTypes.resetWarningCache()` work.
+      // Solution from: https://github.com/facebook/react/issues/18251#issuecomment-609024557
+      React.PropTypes.checkPropTypes(jsConfig.propTypes, this.props, 'prop', ReactDartComponent2.displayName);
```

Comment:
Two things:
1. can we wrap this in a compile-time check so that this call isn't present in the dev bundle?
```suggestion
      if (process.env.NODE_ENV !== 'production') {
        React.PropTypes.checkPropTypes(jsConfig.propTypes, this.props, 'prop', ReactDartComponent2.displayName);
      }
```
2. Can we update the propTypes tests to verify that the correct displayName is being passed? RIght now we capture all of the information passed into prop validators but [only test some of it](https://github.com/cleandart/react-dart/blob/1d3fea5da67fa46b2e95747e684c1296376fc6e2/test/lifecycle_test.dart#L166)

Reply:
It seems like the displayName is being tested [here](https://github.com/cleandart/react-dart/blob/1d3fea5da67fa46b2e95747e684c1296376fc6e2/test/lifecycle_test.dart#L198)? Do you want me to add that check in both places?

Reply:
Oh good catch! It looks like that test would pass if the name is `ReactDartComponent2`, though, and we wouldn't be able to tell if the wrong name is being passed in.

Now that it's wrapped in that `assertsEnabled()` check, though (which wasn't present when the original test was written), I think that case shouldn't happen, and we could update it like so:
```diff
-'componentName': anyOf('PropTypesTestComponent', 'ReactDartComponent2'),
+'componentName': 'PropTypesTestComponent',
```

and then we'd know for sure that the test name is being passed through.

Would you mind updating that test so that we get full coverage on this code?

Reply:
Updated in 2978197!